src/vanta_ledger/services/user_service.py

```
@@ -0,0 +1,247 @@
+            logger.info(f"Created new user: {user_data.username}")
+            return db_user
+            
+        except HTTPException:
+            raise
+        except IntegrityError as e:
+            self.db.rollback()
+            logger.error(f"Database integrity error creating user: {str(e)}")
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="User creation failed - duplicate data"
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)